src/components/Board.js

```diff
@@ -10,24 +10,110 @@ import CARD_DATA from '../data/card-data.json';
 class Board extends Component {
   constructor() {
     super();
-
     this.state = {
       cards: [],
+      boardName: null
     };
   }
 
+  componentDidMount() {
+    this.props.updateStatusCallback(`Loading board: ${this.props.boardName}`, 'success');
+    axios.get(`${this.props.url}${this.props.boardName}/cards`)
+      .then((response) => {
+        this.props.updateStatusCallback(`Board "${this.props.boardName}" successfully loaded` , 'success');
+        this.setState({
+          cards: response.data,
+          boardName: this.props.boardName
+        });
+      })
+
+      .catch((error) => {
+        this.props.updateStatusCallback(error.message, 'error');
+      });
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    // debugger
+    if(prevProps.boardName !== this.props.boardName) {
+      this.props.updateStatusCallback(`Loading board: ${this.props.boardName}`, 'success');
+      axios.get(`${this.props.url}${this.props.boardName}/cards`)
+        .then((response) => {
+          this.props.updateStatusCallback(`Board "${this.props.boardName}" successfully loaded` , 'success');
+          this.setState({
+            cards: response.data,
+            boardName: this.props.boardName
+          });
+        })
+
+      .catch((error) => {
+        this.props.updateStatusCallback(error.message, 'error');
+      });
+    }
+  }
+
+  addCard = (newCard) => {
+    axios.post(`${this.props.url}${this.props.boardName}/cards`, newCard)
+    .then((response) => {
+      this.props.updateStatusCallback('New card successfully added', 'success');
+
+      let updatedCards = this.state.cards;
+      updatedCards.push({
+        card: response.data.card
+      });
+
+      this.setState({
+        cards: updatedCards
+      });
+    })
+    .catch((error) => {
+      this.props.updateStatusCallback('Card could not be added', 'error');
+    });
+  }
+
+  deleteCard = (id) => {
+    axios.delete(`${this.props.url}${this.props.boardName}/cards/${id}`)
+      .then(() => {
+        let updatedCards = this.state.cards;
+        updatedCards.splice(id, 1);
+        this.setState({
+          cards: updatedCards
+        });
+        window.location.reload();
+      })
+      .catch((error) => {
+        this.props.updateStatusCallback(`Card could not be deleted: ${error.message}`, 'error');
+      });
+  }
+
   render() {
+    const cards = this.state.cards.map((card, index) => {
+      return (
+        <div>
```

Comment:
Since this function is returning a collection of Cards, you need to include a key attribute.

Something like:

```jsx
<div key={index}>
```